src/components/video-controller.js

```diff
@@ -66,91 +70,102 @@ export class VideoController {
+    }
+
+    startVideo(videoStream, showControlBar) {
+        this.stopOldVideo(videoStream);
+
+        const isFirstStart = !!videoStream;
+        if (videoStream) {
+            this.videoStream = videoStream;
+        } else {
+            videoStream = this.videoStream;
+        }
+        if (!videoStream) return;
+
+        if (isFirstStart) {
+            this.setAdPlaylist(videoStream.vmap);
+        }
+
+        const firstAdBlock = this.adPlaylist[0];
+        if (firstAdBlock && firstAdBlock.startTime <= 0) {
+            // If we have a preroll, show it immediately, since otherwise it takes a while for the video to load
+            const started = this.startAd(firstAdBlock);
+            if (started) return;
+        }
 
         this.showLoadingSpinner(true);
 
-        this.setAdPlaylist(videoStream.vmap);
+        // Put the video underneath any control overlays.
+        const video = document.createElement('video');
+        this.video = video;
+        this.videoOwner.insertBefore(this.video, this.videoOwner.firstChild);
 
-        const video = this.video;
+        video.src = videoStream.url;
+        video.addEventListener('playing', this.onVideoStarted);
+        video.addEventListener("timeupdate", this.onVideoTimeUpdate);
 
-        const initialVideoTime = Math.max(0, this.currVideoTime || video.currentTime || 0);
+        const initialVideoTime = Math.max(0, this.initialVideoTime || 0);
         this.initialVideoTime = initialVideoTime;
         console.log(`starting video: ${videoStream.title} 
     src: ${videoStream.url}
     at time: ${this.timeDebugDisplay(initialVideoTime)}`);
 
-        video.src = videoStream.url;
-        video.addEventListener('playing', this.onVideoPlaying);
-
+        this.videoStarted = false; // set to true on the first playing event
+        this.currVideoTime = initialVideoTime; // will be updated as video progresses
         video.currentTime = initialVideoTime;
         this.play();
-        this.showControlBar(true);
 
-        if (this.platform.isPS4) {
-            // Using a timeupdate listener seems to hang playback on the PS4.
-            this._videoTimeupdateInterval = setInterval(this.onVideoTimeUpdate, 500);
+        if (showControlBar) {
+            const forceTimer = true;
+            this.showControlBar(forceTimer);
         } else {
-            video.addEventListener("timeupdate", this.onVideoTimeUpdate);
+            this.hideControlBar();
         }
+    }
 
-        const firstAdBlock = this.adPlaylist[0];
-        if (firstAdBlock && firstAdBlock.startTime <= 0) {
-            // If we have a preroll, show it immediately, since otherwise it takes a while for the video to load
-            this.startAd(firstAdBlock);
+    stopOldVideo(newVideoStream) {
+        if (this.video && newVideoStream) {
+            if (this.videoStream === newVideoStream) {
+                return; // already playing.
```

Comment:
would stopCurrentVideoBeforePlaying(newVideoStream) makes more sense?

right now it reads: stopOldVideo, if this.videoStream === newVideoStream, do nothing and do not stopOldVideo because video's already playing. 
which does not make too much sense without knowing what stopOldVideo is being used for.